lib/node_modules/@stdlib/number/float16/ctor/include/stdlib/number/float16/ctor.h

```diff
@@ -0,0 +1,132 @@
+* limitations under the License.
+*/
+
+#ifndef STDLIB_NUMBER_FLOAT16_CTOR_H
+#define STDLIB_NUMBER_FLOAT16_CTOR_H
+
+#include <stdint.h>
+#include <stdbool.h>
+
+/*
+* If C++, prevent name mangling so that the compiler emits a binary file having undecorated names, thus mirroring the behavior of a C compiler.
+*/
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+* An opaque type definition for a half-precision floating-point number.
+*
+* @example
+* stdlib_float16_t x = stdlib_float16_from_float32( 3.14f );
+*
+* float y = stdlib_float16_to_float32( x );
+* // returns 3.140625f
+*/
+typedef struct {
+	/**
+	* Raw IEEE 754 binary16 bit representation.
+	*/
+	uint16_t bits;
+} stdlib_float16_t;
+
+/**
+* Creates a half-precision floating-point number from a single-precision floating-point number.
+*/
+stdlib_float16_t stdlib_float16_from_float32( const float value );
+
+/**
+* Creates a half-precision floating-point number from a double-precision floating-point number.
+*/
+stdlib_float16_t stdlib_float16_from_float64( const double value );
+
+/**
+* Converts a half-precision floating-point number to single-precision floating-point number.
+*/
+float stdlib_float16_to_float32( const stdlib_float16_t value );
+
+/**
+* Converts a half-precision floating-point number to double-precision floating-point number.
+*/
+double stdlib_float16_to_float64( const stdlib_float16_t value );
+
+/**
+* Converts a signed 8-bit integer to a half-precision floating-point number.
+*/
+stdlib_float16_t stdlib_float16_from_int8( const int8_t value );
+
+/**
+* Converts an unsigned 8-bit integer to a half-precision floating-point number.
+*/
+stdlib_float16_t stdlib_float16_from_uint8( const uint8_t value );
+
+/**
+* Converts a signed 16-bit integer to a half-precision floating-point number.
+*/
+stdlib_float16_t stdlib_float16_from_int16( const int16_t value );
+
+/**
+* Converts an unsigned 16-bit integer to a half-precision floating-point number.
+*/
+stdlib_float16_t stdlib_float16_from_uint16( const uint16_t value );
+
+/**
+* Converts a signed 32-bit integer to a half-precision floating-point number.
+*/
+stdlib_float16_t stdlib_float16_from_int32( const int32_t value );
+
+/**
+* Converts an unsigned 32-bit integer to a half-precision floating-point number.
+*/
+stdlib_float16_t stdlib_float16_from_uint32( const uint32_t value );
+
+/**
+* Converts a signed 64-bit integer to a half-precision floating-point number.
+*/
+stdlib_float16_t stdlib_float16_from_int64( const int64_t value );
+
+/**
+* Converts an unsigned 64-bit integer to a half-precision floating-point number.
+*/
+stdlib_float16_t stdlib_float16_from_uint64( const uint64_t value );
+
+/**
+* Converts a half-precision floating-point number to a signed 32-bit integer.
+*/
+int32_t stdlib_float16_to_int32( const stdlib_float16_t value );
+
+/**
+* Converts a half-precision floating-point number to an unsigned 32-bit integer.
+*/
+uint32_t stdlib_float16_to_uint32( const stdlib_float16_t value );
```

Comment:
I don't believe either of these functions are needed. Converting to and from words will be done in the respective packages. And more generally, we don't know to explicitly support lossy operations (e.g., in this case decimal truncation).